src/cryptography/hazmat/backends/openssl/backend.py

```diff
@@ -686,6 +776,68 @@ def cmac_algorithm_supported(self, algorithm):
     def create_cmac_ctx(self, algorithm):
         return _CMACContext(self, algorithm)
 
+    def create_x509_csr(self, builder, private_key, algorithm):
+        # TODO: check type of private key parameter.
```

Comment:
We should add tests for EC and DSA. Those tests will fail until the evp_pkey change is merged though.



> On Jun 18, 2015, at 4:53 PM, André Caron <notifications@github.com> wrote:
> 
> In src/cryptography/hazmat/backends/openssl/backend.py:
> 
> > @@ -686,6 +776,68 @@ def cmac_algorithm_supported(self, algorithm):
> >      def create_cmac_ctx(self, algorithm):
> >          return _CMACContext(self, algorithm)
> >  
> > +    def create_x509_csr(self, builder, private_key, algorithm):
> > +        # TODO: check type of private key parameter.
> Are you saying this PR should wait until another refactoring occurs or should we add the tests to make the EC and DSA cases work in this PR first?
> 
> —
> Reply to this email directly or view it on GitHub.
>